crates/cranelift/src/func_environ.rs

```diff
@@ -2334,18 +2335,23 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
         let mut values = vec![];
         if valtypes.len() == 0 {
             // OK
-        } else if valtypes.len() == 1 {
-            let offset = i32::try_from(self.offsets.vmctx_typed_continuations_payloads()).unwrap();
-            let val = builder
-                .ins()
-                .load(convert_type(valtypes[0]), memflags, base_addr, offset);
-            values.push(val)
+        } else if valtypes.len() <= MAXIMUM_CONTINUATION_PAYLOAD_COUNT as usize {
+            let mut offset =
+                i32::try_from(self.offsets.vmctx_typed_continuations_payloads()).unwrap();
+            for valtype in valtypes {
+                let val = builder
+                    .ins()
+                    .load(convert_type(*valtype), memflags, base_addr, offset);
+                values.push(val);
+                offset += self.offsets.ptr.size_of_vmglobal_definition() as i32;
```

Comment:
Because the size of a global definition is guaranteed to be that of the largest value (i.e., 16 bytes), as per the comment on the function. There doesn't seem to be another dedicated constant with that meaning, but we could define one.

Reply:
Alternatively we can define an alias of it! It may be clearer for our future selves

Reply:
done